main.go

```diff
@@ -46,72 +56,76 @@ func main() {
 	if err != nil {
 		log.Println(err)
 	}
-	var outputString string
-	var auditReport AuditReportBody
 	switch *action {
 	case "audit":
 
 		// minimal using defaults
 
 		// image: extensions/npm-audit:stable
 		// action: audit
-		// level: critical
 
 		// audit repo
 		log.Printf("Auditing repo...\n")
 		auditArgs := []string{
 			"audit",
 			"--json",
 		}
-		outputString = runCommandGetOutput("npm", auditArgs)
-		auditReport = readAuditReport(outputString)
+		reportJson, err := runCommand("npm", auditArgs)
+		if err != nil {
+			if reportJson == "" {
+				log.Fatal(err)
+			}
+			log.Println(err)
+		}
+
+		auditReport := readAuditReport(reportJson)
 
 		log.Printf("Checking for %v vulnerabilities on production repositories\n", prodVulnLevel.String())
 		log.Printf("Checking for %v vulnerabilities on dev dependencies repositories\n", devVulnLevel.String())
-		failBuild := checkIfBuildShouldFail(auditReport, prodVulnLevel, devVulnLevel)
-		if failBuild {
-			auditArgs := []string{
+		failBuild, hasVulns := checkVulnerabilities(auditReport, prodVulnLevel, devVulnLevel)
+
+		if hasVulns {
+			auditArgs = []string{
 				"audit",
 			}
-			// TODO: send report via Slack
-			runCommand("npm", auditArgs)
+			reportString, err := runCommand("npm", auditArgs)
+
+			log.Println(reportString)
+
+			// also send report via Slack
+			if slackEnabled {
+				title := "Vulnerabilities found in your repository."
+				// split on comma and loop through channels
+				channels := strings.Split(*slackChannels, ",")
+				for i := range channels {
+					err := slackWebhookClient.SendMessage(channels[i], title, reportString)
+					if err != nil {
+						log.Printf("Sending status to Slack failed: %v", err)
+					}
+				}
+			}
+			if failBuild {
+				log.Fatal(err)
+			} else {
+				log.Println(err)
+			}
 		} else {
-			// TODO: send report via Slack
-			log.Println("Auditing passed, but you might still have vulnerabilities.")
+			log.Println("No vulnerabilities in your repository for now. Cheers!")
 		}
 	default:
 		log.Fatal("Set `action: <action>` on this step to audit.")
```

Comment:
Perhaps repeat allowed values for `<action>`.

Reply:
But action for now it can only be `audit`